Automattic/MobilePay-Apple
# Project Structure

Changes the project structure while:

* Keeping SPM support
* Adding Cocoapods support

A brief summary of the project organization:

* `Source`: all lib source files should go here
* `Tests`: all lib tests should be added here
* `Example`: the lib examples. As for now, just an iOS example.

### To test

First, open the workspace. It contains both the iOS example and the library. Then:

1. Build the framework
2. Test the framework
3. Run the app and test it

Repeat the process with the `MobilePay-Apple.xcodeproj` and `MobilePayKit-iOS.xcodeproj` alone, they should run without any issues.

### Testing SPM

1. Open `MobilePayKit-iOS.xcodeproj` (inside `Example/iOS`)
2. Delete the `MobilePay-Apple.xcodeproj` (remove reference)
3. Product > Clean Build Folder
4. Go to Swift Packages > Add Package Dependency
5. Add the local path to the lib (eg.: `file:///Users/leandroalonso/Documents/MobilePay-Apple/`)
6. Select "branch" and then finish the process
7. Run the app and make sure everything works

### Testing Cocoapods

First, revert the state of the entire repository to undo the SPM changes. Then:

1. Open `MobilePayKit-iOS.xcodeproj` (inside `Example/iOS`)
2. Delete the `MobilePay-Apple.xcodeproj` (remove reference)
3. Product > Clean Build Folder
4. In the terminal go to `Example/iOS`
5. `pod init`
6. Add `pod 'MobilePayKit', :path => '../..'` to the `Podfile`
7. Open the workspace
8. Run the app and make sure it works
